orb-info/src/orb_token.rs

```diff
@@ -0,0 +1,203 @@
+        let (send, recv) = watch::channel(initial_value);
+        let auth_token_proxy = auth_token_proxy.clone();
+        let cancel_token = cancel_token.clone();
+        let join_handle =
+            tokio::task::spawn(Self::task_inner(auth_token_proxy, send, cancel_token));
+        Ok((join_handle, recv))
+    }
+
+    async fn task_inner(
+        auth_token_proxy: AuthTokenManagerProxy<'_>,
+        send: watch::Sender<String>,
+        cancel_token: CancellationToken,
+    ) -> Result<()> {
+        let token_updater_fut = async move {
+            let mut token_changed = auth_token_proxy.receive_token_changed().await;
+            while let Some(token) = token_changed.next().await {
+                let token = token
+                    .get()
+                    .await
+                    .wrap_err("failed to get token over dbus")?;
+                if send.send(token).is_err() {
+                    // normal for this to fail if for example, all watchers are dropped
+                    return Ok(());
+                }
+            }
+            Ok(())
+        };
+
+        tokio::select! {
+            _ = cancel_token.cancelled() => {
+                debug!("Cancelling token watcher task");
+                Ok(())
+            }
+            result = token_updater_fut => result,
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+    use orb_attest_dbus::AuthTokenManagerT;
+    use std::{
+        sync::{Arc, Mutex},
+        time::Duration,
+    };
+    use zbus::ConnectionBuilder;
+
+    type AuthTokenManagerIface = orb_attest_dbus::AuthTokenManager<Mocked>;
+
+    #[derive(Clone, Debug)]
+    struct Mocked {
+        token: Arc<Mutex<String>>,
+    }
+
+    // Note how we are simply implementing a trait from orb-attest-dbus instead of creating an entirely new struct with zbus macros.
+    // This ensures that the function signatures all match up and we get good compile errors and LSP support.
+    impl AuthTokenManagerT for Mocked {
+        fn token(&self) -> zbus::fdo::Result<String> {
+            Ok(self.token.lock().unwrap().clone())
+        }
+
+        fn force_token_refresh(&mut self, _ctxt: zbus::SignalContext<'_>) {
+            // no-op
+        }
+    }
+
+    // using `dbus_launch` ensures that all tests use their own isolated dbus, and that they can't influence each other.
+    async fn start_dbus_daemon() -> dbus_launch::Daemon {
+        tokio::task::spawn_blocking(|| {
+            dbus_launch::Launcher::daemon()
+                .launch()
+                .expect("failed to launch dbus-daemon")
+        })
+        .await
+        .expect("task panicked")
+    }
+
+    async fn setup_test_server() -> Result<(Connection, dbus_launch::Daemon, Mocked)> {
+        let token = Arc::new(Mutex::new("test_token".to_string()));
+        let mock_manager = Mocked { token };
+        let daemon = start_dbus_daemon().await;
+
+        let connection = ConnectionBuilder::session()?
```

Comment:
```suggestion
        let connection = ConnectionBuilder::address(daemon.address())?
```
without this, dbus uses the default system dbus. Thats probably why the tests were requiring serial as well as the `brew` dbus stuff.

Remove any serial attributes on tests unless it is intending to mutate environment variables (which orb-token should not).

Also, don't forget to stop dropping the daemon (don't assign to `_`) because that will kill the daemon.

Reply:
I did not know that `_var` would keep it alive. Thought it was scoped by usage. Nice!